### PR TITLE
Add footer

### PR DESCRIPTION
**Pull request by YingLiang2**
Added footer with GitHub link

image

### CODE REVIEW

File: web/src/app/components/PageLayout/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"use client"
+
+export default function Footer() {
+    return (
+        <div className="flex px-6 py-4 mb-2">
+            <nav className="flex justify-between w-full">
+                <a href="https://github.com/YingLiang2/Battletype" target="_blank" rel="noopener noreferrer">
```
Comment (jonathanplacatka) [inline web/src/app/components/PageLayout/Footer.tsx]: can you link official repo instead of your fork

File: web/src/app/layout.tsx [hunk 2]
```diff
@@ -15,16 +16,19 @@ export const metadata: Metadata = {
 export default function RootLayout({children,}: Readonly<{children: React.ReactNode;}>) {
   return (
     <html lang="en">
-      <head>
-        <ColorSchemeScript defaultColorScheme="dark"/>
-      </head>
+		<head>
+			<ColorSchemeScript defaultColorScheme="dark"/>
```
Comment (jonathanplacatka) [inline web/src/app/layout.tsx:20]: is this a whitespace mistake? or just a github issue

File: web/src/app/page.tsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ export default function Home() {
 	const router = useRouter();
 
 	return (
-		<main className='flex justify-center my-10 '>
```
Comment (jonathanplacatka) [inline web/src/app/page.tsx:13]: why did you change this to a div?

File: web/src/app/components/PageLayout/Footer.tsx (hunk 1 above)
Comment (YingLiang2) [reply web/src/app/components/PageLayout/Footer.tsx]: Mine is the official repo, because it has the deployment

File: web/src/app/layout.tsx (hunk 2 above)
Comment (YingLiang2) [reply web/src/app/layout.tsx:20]: github issue

File: web/src/app/page.tsx (hunk 3 above)
Comment (YingLiang2) [reply web/src/app/page.tsx:13]: Because the main is now in the layout.tsx

File: web/src/app/components/PageLayout/Footer.tsx (hunk 1 above)
Comment (jonathanplacatka) [reply web/src/app/components/PageLayout/Footer.tsx]: lol what

Comment (YingLiang2) [reply web/src/app/components/PageLayout/Footer.tsx]: like you can see all the frontend deployments on mine

File: web/src/app/page.tsx (hunk 3 above)
Comment (jonathanplacatka) [reply web/src/app/page.tsx:13]: I dont see main in layout.tsx

File: web/src/app/layout.tsx [hunk 4]
```diff
@@ -15,16 +16,19 @@ export const metadata: Metadata = {
...
-      <body className={inter.className}>
-        <MantineProvider defaultColorScheme="dark">
-          <Header />
-          {children}
-        </MantineProvider>
-      </body>
+      <body className={`${inter.className} flex flex-col min-h-screen`}  >
+			<MantineProvider defaultColorScheme="dark">
+				<Header/>
+					<main className="flex-grow">
```
Comment (YingLiang2) [inline web/src/app/layout.tsx:26]: @jonathanplacatka here is where I changed main

File: web/src/app/components/PageLayout/Footer.tsx (hunk 1 above)
Comment (jonathanplacatka) [reply web/src/app/components/PageLayout/Footer.tsx]: We shouldn't really have our deployments in separate versions of the repo anyway. If anything we should move both deployments to the main repo so that server and client versions don't get out of sync.

Long term, the web app and game server should probably go on the same hosting provider so that 1 deployment = 1 version of the website. 

Either way, it's really weird to have the actual website link to a forked version of the main repo

File: web/src/app/layout.tsx (hunk 4 above)
Comment (jonathanplacatka) [reply web/src/app/layout.tsx:26]: oh I see makes sense

File: web/src/app/components/PageLayout/Footer.tsx (hunk 1 above)
Comment (YingLiang2) [reply web/src/app/components/PageLayout/Footer.tsx]: Not really, we'll be putting our githubs on our resume so its not a big chance they'll click it on the main website.

We could do 1 deployment of website, but separatation of concerns is also good because what if vercel or heroku goes down.

Comment (jonathanplacatka) [reply web/src/app/components/PageLayout/Footer.tsx]: If vercel or heroku goes down, then the website won't work anyway, so that's not really a benefit. Moving them to the same hosting provider isn't really a big priority to me though.

however, it's definitely better to put the deployment of both vercel and heroku on the main repo. If your fork wasn't up to date, then you'd be deploying an old version of the website. 

why not just have one repo that is the source of truth for our whole project? you can link your own github on your resume if you want

Comment (YingLiang2) [reply web/src/app/components/PageLayout/Footer.tsx]: My fork won't be out of date anyway, I can always setup a github action to sync it automatically.

**Review by jonathanplacatka — APPROVED** [review]
